sdk/ai/azure-ai-inference/azure/ai/inference/aio/_patch.py

```diff
@@ -0,0 +1,963 @@
+
+        content_type: Optional[str] = kwargs.pop("content_type", _headers.pop("Content-Type", None))
+
+        if body is _Unset:
+            if messages is _Unset:
+                raise TypeError("missing required argument: messages")
+            body = {
+                "extras": extras,
+                "frequency_penalty": frequency_penalty,
+                "max_tokens": max_tokens,
+                "messages": messages,
+                "presence_penalty": presence_penalty,
+                "response_format": response_format,
+                "seed": seed,
+                "stop": stop,
+                "stream": stream,
+                "temperature": temperature,
+                "tool_choice": tool_choice,
+                "tools": tools,
+                "top_p": top_p,
+            }
+            if model_extras is not None and bool(model_extras):
+                body.update(model_extras)
+                _unknown_params = _models._enums.UnknownParams.PASS_THROUGH  # pylint: disable=protected-access
+            body = {k: v for k, v in body.items() if v is not None}
+        elif isinstance(body, dict) and "stream" in body and isinstance(body["stream"], bool):
+            stream = body["stream"]
+        content_type = content_type or "application/json"
+        _content = None
+        if isinstance(body, (IOBase, bytes)):
+            _content = body
+        else:
+            _content = json.dumps(body, cls=SdkJSONEncoder, exclude_readonly=True)  # type: ignore
+
+        _request = build_chat_completions_complete_request(
+            unknown_params=_unknown_params,
+            content_type=content_type,
+            api_version=self._config.api_version,
+            content=_content,
+            headers=_headers,
+            params=_params,
+        )
+        path_format_arguments = {
+            "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
+        }
+        _request.url = self._client.format_url(_request.url, **path_format_arguments)
+
+        _stream = stream or False
+        pipeline_response: PipelineResponse = await self._client._pipeline.run(  # type: ignore # pylint: disable=protected-access
+            _request, stream=_stream, **kwargs
+        )
+
+        response = pipeline_response.http_response
+
+        if response.status_code not in [200]:
+            if _stream:
+                await response.read()  # Load the body in memory and close the socket
+            map_error(status_code=response.status_code, response=response, error_map=error_map)
+            raise HttpResponseError(response=response)
+
+        if _stream:
+            return _models.AsyncStreamingChatCompletions(response)
+
+        return _deserialize(_models.ChatCompletions, response.json())  # pylint: disable=protected-access
+
+    # Cache here the results of get_model_info call
+    _model_info: Optional[_models.ModelInfo] = None
```

Comment:
This is a class-level attribute. I don't think that is what you intended?

Reply:
Good catch! I was doing C# coding in Python... I removed the statement `_model_info: Optional[_models.ModelInfo] = None`. And updated the methods `get_model_info()` and `__str__()`. As of now, if you create a client directly, it will no have the attribute `_model_info`. Only after calling `get_model_info()`, the attribute `_model_info` will be created and will hold the model info results from the service call. The alternative is to override the `__init__` method in `_patch.py` for all the clients (sync and async clients), and define plus initialize `_model_info` to `None` there. At this point (just to support _model_info), I don't want to override all `__init__` methods.